packages/flutter/lib/src/widgets/view.dart

```diff
@@ -106,13 +142,578 @@ class View extends StatelessWidget {
+      _effectivePipelineOwner.dispose();
+    }
+    super.unmount();
+  }
 }
 
 class _ViewScope extends InheritedWidget {
   const _ViewScope({required this.view, required super.child});
 
-  final FlutterView view;
+  final FlutterView? view;
 
   @override
   bool updateShouldNotify(_ViewScope oldWidget) => view != oldWidget.view;
 }
+
+class _PipelineOwnerScope extends InheritedWidget {
+  const _PipelineOwnerScope({
+    required this.pipelineOwner,
+    required super.child,
+  });
+
+  final PipelineOwner pipelineOwner;
+
+  @override
+  bool updateShouldNotify(_PipelineOwnerScope oldWidget) => pipelineOwner != oldWidget.pipelineOwner;
+}
+
+class _MultiChildComponentWidget extends Widget {
+  const _MultiChildComponentWidget({
+    super.key,
+    List<Widget> views = const <Widget>[],
+    Widget? child,
+  }) : _views = views, _child = child;
+
+  // It is up to the subclasses to make the relevant properties public.
+  final List<Widget> _views;
+  final Widget? _child;
+
+  @override
+  Element createElement() => _MultiChildComponentElement(this);
+}
+
+/// A collection of sibling [View]s.
+///
+/// This widget can only be used in places were a [View] widget is allowed. In
+/// practical terms, it can be used at the root of the widget tree outside of
+/// any [View] widget, as a child to a another [ViewCollection], or in the
+/// [ViewAnchor.view] slot of a [ViewAnchor] widget. It is not required to be a
+/// direct child of those widgets; other non-[RenderObjectWidget]s may appear
+/// in between the two (such as an [InheritedWidget]).
+///
+/// Similarly, the [views] children of this widget must be [View]s, but they
+/// may be wrapped in additional non-[RenderObjectWidget]s (e.g.
+/// [InheritedWidget]s).
+class ViewCollection extends _MultiChildComponentWidget {
+  /// Creates a [ViewCollection] widget.
+  ///
+  /// The provided list of [views] must contain at least one widget.
+  const ViewCollection({super.key, required super.views}) : assert(views.length > 0);
+
+  /// The [View] descendants of this widget.
+  ///
+  /// The [View]s may be wrapped in other non-[RenderObjectWidget]s (e.g.
+  /// [InheritedWidget]s). However, no [RenderObjectWidget] is allowed to appear
+  /// between the [ViewCollection] and the next [View] widget.
+  List<Widget> get views => _views;
+}
+
+/// Decorates a [child] widget in a surrounding [View] with a side-[View].
+///
+/// This widget must have a [View] ancestor, into which the [child] widget
+/// is rendered.
+///
+/// Typically, a [View] or [ViewCollection] widget is used in the [view] slot to
+/// define the content of the side view(s). Those widgets may be wrapped in
+/// other non-[RenderObjectWidget]s (e.g. [InheritedWidget]s). However, no
+/// [RenderObjectWidget] is allowed to appear between the [ViewAnchor] and the
+/// next [View] widget in the [view] slot. The widgets in the [view] slot have
+/// access to all [InheritedWidget]s above the [ViewAnchor] in the tree.
```

Comment:
```suggestion
/// [ViewAnchor] is the core of using multiple views in a Flutter app.  The
/// [view] parameter branches out a new view, which does not affect [child], and
/// has access to the widget tree information up to this point, such as
/// [InheritedWidget] states.
///
/// The [view] argument should a [View] widget, or a [View] widget wrapped in
/// other non-[RenderObjectWidget] widgets. Common examples of such widgets are:
///
///   * [InheritedWidget],
///   * [ViewCollection], which allows fitting multiple [View] widgets in a
///     single [view] slot.
///
/// Having a [RenderObjectWidget] between the [ViewAnchor] argument and a [View]
/// widget triggers a runtime error.
```

Reply:
I'm proposing this because:
* If I understand correctly, `ViewAnchor` is designed to be the only entry that allows injecting a `View` into a widget tree, unless the developer writes their own version of `ViewAnchor`. It might be better to explain this explicitly.
* It's easier to understand to sort `ViewCollection` into the same category as `InheritedWidget` as "intermediate non-ROWidgets", so that it's clearer that the "no ROW zone" is defined by `view` and `View`.

Reply:
I added some more color to this doc to link it to the different zones of the widget tree.